### DISCUSSION TITLE
Put file locations of the actual build into the documentation?

### DISCUSSION THREAD

**Discussion by ndim** [avrdudes/avrdude#1628] [Ideas]
The documentation avrdude installs (both the `avrdude.1` man page and the `avrdude.texi` derived info page, avrdude.pdf, and HTML docs) contain references to file locations, e.g. to `avrdude.conf` and `avrdude.pdf`.

Would you be interested in having the generated documentation contain the exact location this specific build uses, in addition to the existing description of the general locations not specific to this build? That could be helpful at least for Unix type builds. For Windows and MacOS, maybe less.

This would need adaptation of the some parts of the docs, e.g. the section on where avrdude looks for `avrdude.conf`. I am not even certain the text documentation actually 100% describes what `src/main.c` actually does.

If there is general interest in this, we can convert this to an issue and I could prepare a PR which adapts the documentation text in an appropriate way, and then changes both the `cmake` and the `automake` based build systems to generate the docs with the proper file locations substituted in the text.

That would save me a bit of work in the Fedora package, and I have seen the Debian package patch file locations as well.

**Comment by stefanrueger** [comment]
Thanks for noticing! I am at sixes and sevens over the suggestion: On the one hand it is really neat to have the correct location in the doc for that particular instance, but on the other hand I fear that a hosted version of the html documentation or .pdf is accessed by users from all sort of OSes and compiled instances, which might confuse them as the documented location would be for a different system. It would be great to update the documentation to reflect what main.c actually does (if it does not already) in generic terms, and it might be useful to have the location of the specific build as an *example* clearly stating for which OS that holds. If that's what you suggest we could consider this. There might be voices against adding a specific build-dependent example to the docs, eg, as this is a proliferation of different doc versions for the same release. So, let's hear from others as well

**Comment by ndim** [comment]
OK, I will start with reading through all mentioned file locations in the documentation and adapt that to accurately describe what

  * the compiled `avrdude` executable
  * the cmake and the automake based build systems' installation

actually do.

See https://github.com/ndim/avrdude/tree/file-locations-in-docs and the corresponding https://github.com/avrdudes/avrdude/pull/1637 (currently Draft) PR.
